Sources/MapboxNavigation/NavigationMapView+VanishingRouteLine.swift

```diff
@@ -202,34 +202,55 @@ extension NavigationMapView {
         if traveledDifference == 0.0 {
             return
         }
-        let startDate = Date()
-        vanishingRouteLineUpdateTimer = Timer.scheduledTimer(withTimeInterval: 0.05, repeats: true, block: { timer in
-            let timePassedInMilliseconds = Date().timeIntervalSince(startDate) * 1000
-            if timePassedInMilliseconds >= 980 {
-                timer.invalidate()
-                return
-            }
-            
-            let newFractionTraveled = self.preFractionTraveled + traveledDifference * timePassedInMilliseconds.truncatingRemainder(dividingBy: 1000) / 1000
-            
+        switch puckType {
+        case .courseView(configuration: _):
+            let startDate = Date()
+            vanishingRouteLineUpdateTimer = Timer.scheduledTimer(withTimeInterval: 0.05, repeats: true, block: { timer in
+                let timePassedInMilliseconds = Date().timeIntervalSince(startDate) * 1000
+                // In case if the time was already in the last interval - invalidate the timer and wait for the next routeProgress update.
+                if timePassedInMilliseconds >= 980 {
+                    timer.invalidate()
+                    return
+                }
+                
+                let newFractionTraveled = self.preFractionTraveled + traveledDifference * timePassedInMilliseconds.truncatingRemainder(dividingBy: 1000) / 1000
+                
+                do {
+                    try self.mapView.mapboxMap.style.updateLayer(withId: mainRouteLayerIdentifier) { (lineLayer: inout LineLayer) throws in
+                        let congestionSegments = routeProgress.route.congestionFeatures(legIndex: self.currentLegIndex, roadClassesWithOverriddenCongestionLevels: self.roadClassesWithOverriddenCongestionLevels)
+                        let mainRouteLayerGradient = self.routeLineGradient(congestionSegments,
+                                                                            fractionTraveled: newFractionTraveled)
+                        
+                        lineLayer.lineGradient = .expression(Expression.routeLineGradientExpression(mainRouteLayerGradient))
+                    }
+                    
+                    try self.mapView.mapboxMap.style.updateLayer(withId: mainRouteCasingLayerIdentifier) { (lineLayer: inout LineLayer) throws in
+                        let mainRouteCasingLayerGradient = self.routeLineGradient(fractionTraveled: newFractionTraveled)
+                        
+                        lineLayer.lineGradient = .expression(Expression.routeLineGradientExpression(mainRouteCasingLayerGradient))
+                    }
+                } catch {
+                    print("Failed to update main route line layer.")
+                }
```

Comment:
This is basically a duplicate of code in `.default` case (other than `newFractionTraveled` and `fractionTraveled` properties usage). Can it be extracted into separate method?

Reply:
Thanks for the suggestion, I'll update the implementation of the vanishing route line.

Reply:
Due to the [comment](https://github.com/mapbox/mapbox-navigation-ios/pull/2968#issuecomment-857285377) below, the vanishing route line method now is still using the Timer incase of overwhelming calculation.